lib/monday-api-client/monday-api-client.ts

```diff
@@ -0,0 +1,78 @@
+import {BadRequestError, InternalServerError} from 'errors/apps-sdk-error';
+import {Token} from 'types/general';
+import {isDefined} from 'types/guards';
+import {
+    APIData,
+    APIFetchResponse,
+    APIOptions,
+    APIResponse,
+    APIVersion,
+    IMondayApiClient
+} from 'types/monday-api-client';
+import {fetchWrapper} from 'utils/fetch-wrapper';
+import { Logger } from 'utils/logger';
+const logger = new Logger('MondayApiClient', { mondayInternal: true });
+const API_URL = 'https://api.monday.com/v2/';
+export class MondayApiClient implements IMondayApiClient {
+    private token?: Token;
+
+    private apiVersion?: APIVersion;
+
+    constructor(token?: Token) {
+        this.token = token;
+    }
+
+    setToken(token: Token) {
+        this.token = token;
+    }
+
+    setApiVersion(version: APIVersion){
+        this.apiVersion = version
+    }
+
+    async api(query: string, options: APIOptions = {}) {
+        if (!this.token) {
+            logger.warn('[mondayApiClientFetch] Error Token is missing');
+            throw new BadRequestError('Token is missing');
+        }
+
+        const params: APIData = { query, variables: options.variables };
```

Comment:
Having variables as undefined is ok? Should we transfer it if this is the scenario?